src/repository/stockRepository.js

```diff
@@ -10,11 +11,59 @@ export default class StockRepository {
         var recordset = await request.query(`SELECT sm.Ticker, sm.CompanyName, u.FirstName, u.LastName, u.EmailAddress
                         FROM StockMetadata sm (NOLOCK)
                         INNER JOIN UserStock us (NOLOCK)
-                        ON us.StockID = sm.StockID
+                        ON us.Ticker = sm.Ticker
                         INNER JOIN [User] u (NOLOCK)
-                        ON u.UserId = us.UserID`);
+                        ON u.Username = us.Username`);
             
         var stocksFromDb = recordset.recordsets[0];
         return stocksFromDb;
     }
+
+    async postCandlestickData(stockData) {
+        await sql.connect(connectionString);
+        var request = new sql.Request();
+
+        var table = new sql.Table("Candlestick");
+        table.columns.add('Ticker', sql.VarChar(4), {nullable: false});
+        table.columns.add('Timestamp', sql.DateTime2, {nullable: false});
+        table.columns.add('HighPrice', sql.Numeric, {nullable: false});
+        table.columns.add('LowPrice', sql.Numeric, {nullable: false});
+        table.columns.add('OpenPrice', sql.Numeric, {nullable: false});
+        table.columns.add('ClosePrice', sql.Numeric, {nullable: false});
+
+        _.forEach(stockData.Candlesticks, (candlestick) => {
+            table.rows.add(
+                stockData.Ticker,
+                candlestick.Timestamp,
+                candlestick.HighPrice,
+                candlestick.LowPrice,
+                candlestick.OpenPrice,
+                candlestick.ClosePrice);
+        });
+
+        await request.bulk(table);
+    }
+
+    async postStockMetadata(ticker, companyName) {
```

Comment:
if the table gets bigger, we need to update the input parameters everything. would it better if we passed in an object. the size of the object can change, but the signature of the method doesn't

Reply:
Done